oozie-to-airflow/converter/primitives.py

```diff
@@ -14,53 +14,72 @@
 # limitations under the License.
 """Class for Airflow relation"""
 from collections import OrderedDict
-from typing import Set, Optional, Dict, NamedTuple, Any
+from typing import Any, Dict, Optional, Set
+
+from orderedset import OrderedSet
 
 from airflow.utils.trigger_rule import TriggerRule
 
-# Pylint and flake8 does not understand forward references
-# https://www.python.org/dev/peps/pep-0484/#forward-references
-from converter import parsed_node  # noqa: F401 pylint: disable=unused-import
 from utils.template_utils import render_template
+from utils.variable_name import convert_to_python_variable
 
 
-class Relation(NamedTuple):
+class Relation:
     """Class for Airflow relation"""
 
-    from_task_id: str
-    to_task_id: str
+    def __init__(self, from_task_id: str, to_task_id: str):
+        self.from_task_id: str = from_task_id
+        self.to_task_id: str = to_task_id
+
+    @property
+    def from_task_variable_name(self):
+        return convert_to_python_variable(self.from_task_id)
```

Comment:
I am trying to separate the process from the rendering process. Here a specific solution is indicated, which may or may not be used in a particular rendering engine. I would prefer to create filterJinja from this

Reply:
I need to understand what you want to do and especially why this solution for now is not good. It does not block us if we want to use a different solution in the future.

Reply:
For now, I think it is acceptable We mix the responsibility of several classes. I would have seen that primitive objects do not contain any logic.

Reply:
I do not know if we understood each other well, so in my free time I prepared a commit.
Pay particular attention to the change in the `oozie-to-airflow/mappers/start_mapper.py` file. I am afraid of mixing the responsibility of one class.